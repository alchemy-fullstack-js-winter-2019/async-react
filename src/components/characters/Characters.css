ul li {
  list-style: none;
  display: inline-block;
  margin: 2vw;
}

.characters {
  float: left;
  width: 65vw;
}

img {
  width: 200px;
}