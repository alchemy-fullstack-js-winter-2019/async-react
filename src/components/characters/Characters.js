import Character from './Character';
import Results from '../paging/Results';
import { getCharacters } from '../../services/rickAndMortyApi';

export default Results(Character, getCharacters);
