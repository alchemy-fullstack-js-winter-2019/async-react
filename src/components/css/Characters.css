ul {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
}

li {
  list-style-type: none;
  position: relative;
  width: 50%;
}

h2 {
  font-size: 1.1em;
  width: 250px;
}