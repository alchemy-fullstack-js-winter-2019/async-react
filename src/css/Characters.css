.characters {
  font-family: sans-serif;
  ul {
    display: flex;
    justify-content: space-around;
    flex-wrap: wrap;
    color: white;
    
    li {
      padding: 20px;
      background: indigo;
      list-style: none;
      border: 5px black;
      width: 200px;
      margin: 10px;
    }
  }
}
