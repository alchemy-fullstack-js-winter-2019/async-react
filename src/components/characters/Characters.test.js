import React from 'react';
import renderer from 'react-test-renderer';
import Characters from './Characters';

jest.mock('../../services/rickAndMortyApi.js');

describe('Characters', () => {
  it('matches a snapshot', () => {
    const tree = renderer.create(
      <Characters />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
