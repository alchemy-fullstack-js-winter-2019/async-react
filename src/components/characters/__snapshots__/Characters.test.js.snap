// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters matches a snapshot 1`] = `
<table>
  <tbody />
</table>
`;
